frontend/viewer/src/lib/components/field-editors/multi-select.svelte

```diff
@@ -0,0 +1,121 @@
+<script lang="ts" generics="Value">
+  import { Badge } from '$lib/components/ui/badge';
+  import { Button } from '$lib/components/ui/button';
+  import { Popover, PopoverContent, PopoverTrigger } from '$lib/components/ui/popover';
+  import { IsMobile } from '$lib/hooks/is-mobile.svelte';
+  import { tick } from 'svelte';
+  import { t } from 'svelte-i18n-lingui';
+  import { Checkbox } from '../ui/checkbox';
+  import { Command, CommandEmpty, CommandGroup, CommandInput, CommandItem, CommandList } from '../ui/command';
+  import { Drawer, DrawerContent, DrawerTrigger } from '../ui/drawer';
+  import { Icon } from '../ui/icon';
+
+  let {
+    values = $bindable(),
+    ...constProps
+  }: {
+    values: Value[];
+    options: Value[];
+    toLabel: (value: Value) => string;
+    placeholder?: string;
+    filterPlaceholder?: string;
+    resultsPlaceholder?: string;
+  } = $props();
+
+  const { options, toLabel, placeholder, filterPlaceholder, resultsPlaceholder } = constProps;
```

Comment:
this doesn't actually make them const though does it? I'm not sure there is a way to prevent changes to props.

Reply:
It's common to do `const { ... } = $props()`, because/so that the component doesn't write to non-bindeable props. But you can't use `const` if you have a bindeable prop, because the component needs to write to it (`values` in this case). By having 2 declarations, I can still define everything else as const.

But this way buggy. I needed `$derived(constProps)`.

Reply:
I see, so this is expressing to the component that these values (eg toLabel) are read only. I think the confusing thing is that I would assume since toLabel is const, that it should never change after the component is created, but that's not true as the component user could change that value after the fact. But I guess that's also the case with `const { ... } = $props()` too.